packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -1020,6 +1020,36 @@ void main() {
     }
   });
 
+  testWidgets('Enable filtering with custom filter callback that filter text case sensitive', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+    await tester.pumpWidget(MaterialApp(
+      theme: themeData,
+      home: Scaffold(
+        body: DropdownMenu<TestMenu>(
+          requestFocusOnTap: true,
+          enableFilter: true,
+          filterCallback: (List<DropdownMenuEntry<TestMenu>>entries,String filter){
+            return entries.where((DropdownMenuEntry<TestMenu> element) => element.label.contains(filter)).toList();
+          },
+          dropdownMenuEntries: menuChildren,
+        ),
+      ),
+    ));
+
+    // Open the menu
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pump();
+
+    await tester.enterText(find
+        .byType(TextField)
+        .first, 'item');
```

Comment:
If we want to test case sensitive, and with input "item", there should be no widget at all right?

Reply:
That's exactly right. I don't really know how to properly test that since `expect(find.widgetWithText(MenuItemButton, menu.label)` returns 1 when the search returns nothing. Do you know a proper way to test it? I followed the above test called: `Enable filtering`

Reply:
We can make the test look like this:
```dart
testWidgets('Enable filtering with custom filter callback that filter text case sensitive', (WidgetTester tester) async {
    final ThemeData themeData = ThemeData();
    final TextEditingController controller = TextEditingController();
    addTearDown(controller.dispose);

    await tester.pumpWidget(MaterialApp(
      theme: themeData,
      home: Scaffold(
        body: DropdownMenu<TestMenu>(
          requestFocusOnTap: true,
          enableFilter: true,
          filterCallback: (List<DropdownMenuEntry<TestMenu>> entries, String filter) {
            return entries.where((DropdownMenuEntry<TestMenu> element) => element.label.contains(filter)).toList();
          },
          dropdownMenuEntries: menuChildren,
          controller: controller,
        ),
      ),
    ));

    // Open the menu
    await tester.tap(find.byType(DropdownMenu<TestMenu>));
    await tester.pump();

    await tester.enterText(find.byType(TextField).first, 'item');
    expect(controller.text, 'item');
    await tester.pumpAndSettle();
    for (final TestMenu menu in TestMenu.values) {
      expect(find.widgetWithText(MenuItemButton, menu.label).hitTestable(), findsNothing);
    }

    await tester.enterText(find.byType(TextField).first, 'Item');
    expect(controller.text, 'Item');
    await tester.pumpAndSettle();
    expect(find.widgetWithText(MenuItemButton, 'Item 0').hitTestable(), findsOneWidget);
    expect(find.widgetWithText(MenuItemButton, 'Item 2').hitTestable(), findsOneWidget);
    expect(find.widgetWithText(MenuItemButton, 'Item 3').hitTestable(), findsOneWidget);
    expect(find.widgetWithText(MenuItemButton, 'Item 4').hitTestable(), findsOneWidget);
    expect(find.widgetWithText(MenuItemButton, 'Item 5').hitTestable(), findsOneWidget);
  });
```

The reason that I added a `hitTestable()` is that `DropdownMenu` layouts children for the first time when the dropdown menu is created to get the size of children but doesn't really paint them nor make them clickable. When we open the menu, we layout the children, paint them and make the children clickable. So in test, using `hitTestable` helps find the correct widget that we want.